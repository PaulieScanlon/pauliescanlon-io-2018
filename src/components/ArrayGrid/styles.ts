import styled from "react-emotion";

import { common } from "../../styles/";

export const ArrayGridWrapper = styled.div({
  label: "array-grid-wrapper",
  display: "flex",
  flexWrap: "wrap",
  margin: `0 -${common.padding / 2}px 0 -${common.padding / 2}px`
});
