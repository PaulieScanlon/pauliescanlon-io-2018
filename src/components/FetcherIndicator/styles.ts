import styled from "react-emotion";

export const IndicatorWrapper = styled.div({
  label: "indicator-wrapper",
  with: "100%",
  minHeight: 300
});
