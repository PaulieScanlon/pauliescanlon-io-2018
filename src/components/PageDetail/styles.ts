import styled from "react-emotion";

export const PageDetailWrapper = styled.div({
  label: "page-detail-wrapper"
});
