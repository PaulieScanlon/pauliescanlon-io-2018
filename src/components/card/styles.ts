import styled from 'react-emotion';

export const CardWrapper = styled.div({
  label: 'card-wrapper'
});
