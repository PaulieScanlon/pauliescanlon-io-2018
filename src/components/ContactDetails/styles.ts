import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";
import { common } from "../../styles/";

export const ContactDetailsWrapper = styled.div({
  label: "contact-details-wrapper",
  display: "flex",
  flexDirection: "column"
});

export const ContactLink = styled.a({
  label: "contact-link",
  cursor: "pointer",
  fontSize: 14,
  color: draculaTheme.comment,
  transition: `all ${common.hoverTransition}s`,
  "&:hover": {
    color: draculaTheme.purple
  }
});
