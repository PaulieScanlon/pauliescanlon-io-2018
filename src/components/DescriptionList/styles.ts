import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";

export const DL = styled.dl({
  label: "descriptions-list"
});

export const DT = styled.dt({
  label: "descriptions-term",
  color: draculaTheme.cyan
});

export const DD = styled.dd({
  label: "descriptions-description",
  margin: "0px 0px 10px 0px",
  color: draculaTheme.comment,
  a: {
    color: `${draculaTheme.comment}!important`
  }
});
