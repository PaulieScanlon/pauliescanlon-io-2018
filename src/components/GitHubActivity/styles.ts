import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";
import { common } from "../../styles/";

export const GitHubActivityWrapper = styled.div({
  label: "git-hub-activity-wrapper",
  display: "flex",
  flexDirection: "column"
});

export const RepoLink = styled.a({
  label: "repo-link",
  cursor: "pointer",
  fontSize: 14,
  color: draculaTheme.comment,
  transition: `all ${common.hoverTransition}s`,
  "&:hover": {
    color: draculaTheme.purple
  }
});
