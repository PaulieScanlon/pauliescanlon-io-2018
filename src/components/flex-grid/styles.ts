import styled from 'react-emotion';

// import { IPx } from '../../styles/breakpoints';
import { mq } from '../../styles/breakpoints';

interface IStyles {
  xs?: number;
  sm?: number;
}

export const FlexContainer = styled.div({
  label: 'flex-container',
  display: 'flex',
  flexWrap: 'wrap'
});

export const FlexItem = styled.div<IStyles>(
  {
    label: 'flex-item',
    position: 'relative',
    boxSizing: 'border-box'
  },
  ({ xs, sm }) => ({
    [mq[0]]: {
      border: '1px solid red',
      width: `calc(${100 / xs}% - ${0}px)`
    },
    [mq[1]]: {
      border: '1px solid blue',
      width: `calc(${100 / sm}% - ${0}px)`
    }
  })
);
