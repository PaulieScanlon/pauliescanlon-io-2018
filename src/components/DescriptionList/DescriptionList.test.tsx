import * as React from "react";
import { shallow, mount } from "enzyme";

import { DescriptionList } from "./";
import { DL, DT, DD } from "./styles";

import { defaultList } from "./data.mock";

const props = {
  data: defaultList
};

const shallowDefault = () => {
  return shallow(<DescriptionList {...props} />);
};

const mountDefault = () => {
  return mount(<DescriptionList {...props} />);
};

describe("<DescriptionList />", () => {
  it("renders correctly", () => {
    const wrapper = shallowDefault();
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it("has required props", () => {
    const wrapper = mountDefault();
    expect(wrapper.props()).toEqual({ ...props });
  });
  it("has a DL", () => {
    const wrapper = shallowDefault();
    const dl = wrapper.find(DL);
    expect(dl).toHaveLength(1);
  });
  it("has correct amount of DT", () => {
    const wrapper = mountDefault();
    const dt = wrapper.find(DT);
    expect(dt.length).toEqual(5);
  });
  it("has correct amount of DD", () => {
    const wrapper = mountDefault();
    const dd = wrapper.find(DD);
    expect(dd.length).toEqual(5);
  });
});
