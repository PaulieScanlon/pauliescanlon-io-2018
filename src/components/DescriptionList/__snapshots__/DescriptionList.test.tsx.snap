// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionList /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DescriptionList
    data={
      Array [
        Object {
          "description": "Ipsum",
          "term": "Lorem",
        },
        Object {
          "description": "Sit",
          "term": "Dolo",
        },
        Object {
          "description": "Adipiscing",
          "term": "Consectetur",
        },
        Object {
          "description": "Vitae",
          "term": "Fusce",
        },
        Object {
          "description": "Suscipit",
          "term": "Molestie",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <React.Fragment>
          <DT>
            Lorem
          </DT>
          <DD>
            Ipsum
          </DD>
        </React.Fragment>,
        <React.Fragment>
          <DT>
            Dolo
          </DT>
          <DD>
            Sit
          </DD>
        </React.Fragment>,
        <React.Fragment>
          <DT>
            Consectetur
          </DT>
          <DD>
            Adipiscing
          </DD>
        </React.Fragment>,
        <React.Fragment>
          <DT>
            Fusce
          </DT>
          <DD>
            Vitae
          </DD>
        </React.Fragment>,
        <React.Fragment>
          <DT>
            Molestie
          </DT>
          <DD>
            Suscipit
          </DD>
        </React.Fragment>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <DT>
              Lorem
            </DT>,
            <DD>
              Ipsum
            </DD>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Lorem",
            },
            "ref": null,
            "rendered": "Lorem",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Ipsum",
            },
            "ref": null,
            "rendered": "Ipsum",
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <DT>
              Dolo
            </DT>,
            <DD>
              Sit
            </DD>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Dolo",
            },
            "ref": null,
            "rendered": "Dolo",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sit",
            },
            "ref": null,
            "rendered": "Sit",
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <DT>
              Consectetur
            </DT>,
            <DD>
              Adipiscing
            </DD>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Consectetur",
            },
            "ref": null,
            "rendered": "Consectetur",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Adipiscing",
            },
            "ref": null,
            "rendered": "Adipiscing",
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <DT>
              Fusce
            </DT>,
            <DD>
              Vitae
            </DD>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Fusce",
            },
            "ref": null,
            "rendered": "Fusce",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Vitae",
            },
            "ref": null,
            "rendered": "Vitae",
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <DT>
              Molestie
            </DT>,
            <DD>
              Suscipit
            </DD>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Molestie",
            },
            "ref": null,
            "rendered": "Molestie",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Suscipit",
            },
            "ref": null,
            "rendered": "Suscipit",
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <React.Fragment>
            <DT>
              Lorem
            </DT>
            <DD>
              Ipsum
            </DD>
          </React.Fragment>,
          <React.Fragment>
            <DT>
              Dolo
            </DT>
            <DD>
              Sit
            </DD>
          </React.Fragment>,
          <React.Fragment>
            <DT>
              Consectetur
            </DT>
            <DD>
              Adipiscing
            </DD>
          </React.Fragment>,
          <React.Fragment>
            <DT>
              Fusce
            </DT>
            <DD>
              Vitae
            </DD>
          </React.Fragment>,
          <React.Fragment>
            <DT>
              Molestie
            </DT>
            <DD>
              Suscipit
            </DD>
          </React.Fragment>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <DT>
                Lorem
              </DT>,
              <DD>
                Ipsum
              </DD>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Lorem",
              },
              "ref": null,
              "rendered": "Lorem",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Ipsum",
              },
              "ref": null,
              "rendered": "Ipsum",
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <DT>
                Dolo
              </DT>,
              <DD>
                Sit
              </DD>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Dolo",
              },
              "ref": null,
              "rendered": "Dolo",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sit",
              },
              "ref": null,
              "rendered": "Sit",
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <DT>
                Consectetur
              </DT>,
              <DD>
                Adipiscing
              </DD>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Consectetur",
              },
              "ref": null,
              "rendered": "Consectetur",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Adipiscing",
              },
              "ref": null,
              "rendered": "Adipiscing",
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <DT>
                Fusce
              </DT>,
              <DD>
                Vitae
              </DD>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Fusce",
              },
              "ref": null,
              "rendered": "Fusce",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Vitae",
              },
              "ref": null,
              "rendered": "Vitae",
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <DT>
                Molestie
              </DT>,
              <DD>
                Suscipit
              </DD>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Molestie",
              },
              "ref": null,
              "rendered": "Molestie",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Suscipit",
              },
              "ref": null,
              "rendered": "Suscipit",
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
